packages/react-ui-core/src/Form/RadioGroup.js

```diff
@@ -73,6 +77,19 @@ export default class RadioGroup extends Component {
     }
   }
 
+  handleClick = event => {
+    // Check if value was already selected, should we unselect it?
+    if (this.props.allowUnselect && this.state.value === event.target.value) {
+      this.setState({ value: null })
+      if (this.props.onUnselect) {
+        this.props.onUnselect(event)
+      }
+    }
+    if (this.props.onClick) {
```

Comment:
This is causing issues.  If something changes, this gets fired twice if I add an onClick prop
```
export const UnselectRadioGroup = (
  <RadioGroup
    name={'radio-group-4'}
    orientation="vertical"
    hideInputElement
    fields={[
      { label: 'Uno', checked: true, value: 'Uno' },
      { label: 'Dos', checked: false, value: 'Dos' },
    ]}
    onChange={event => action('change')(event.target.value)}
    allowUnselect
    onClick={event => action('select')(event.target.value)}
    onUnselect={event => action('unselect')(event.target.value)}
  />
)
```

![screen shot 2018-02-22 at 8 04 11 am](https://user-images.githubusercontent.com/10748767/36549307-54b1d27e-17a7-11e8-9cc3-ea324aa668d4.png)

I don't think `onClick` needs to be added to this component.

Reply:
@tadjohnston Thanks, I added the `onClick` prop because I thought I was "consuming" the onclick event from the RadioButton and needed to pass it through in case someone wanted an `onClick` handler for the RadioGroup, but that doesn't seem to be the case. I will remove.

If you think everything else is okay, I'll move forward with writing tests.

Reply:
Yeah, I figured that's why you did it. I had to 2nd guess myself as well and look at it a couple times. 😄